R/connection.R

```diff
@@ -149,3 +151,25 @@ id <- function(db_table_id, conn = NULL, allow_table_only = TRUE) {
 
   return(DBI::Id(schema = db_schema, table = db_table))
 }
+
+#' @importFrom rlang %||%
+#' @export
+id.tbl_dbi <- function(db_table_id, conn = NULL, allow_table_only = TRUE) {
+  table_ident <- dbplyr::remote_table(db_table_id)
+
+  id <- with(table_ident, {
+    setNames(list(catalog = catalog,
+                  schema = ifelse(is.na(schema), get_schema(conn), schema),
+                  table = table),
+             c("catalog", "schema", "table")) |>
+      (\(.x) subset(.x, !is.na(.x)))() |>
+      do.call(DBI::Id, args = _)
+  })
+
+  if (!is.null(conn) && !identical(conn, dbplyr::remote_con(db_table_id))) {
+    rlang::warn("Table connection is different than conn",
+                frequency = "once")
```

Comment:
Are you sure warning once here is a good idea?

If I make a mistake with table A and again with table B, I will only get a warning for table A and not table B?

Reply:
Good point; I must have been haunted by the "`consider using in_schema()`" messages.

It is definitely not transparent to show this warning once across all cases, and this behaviour will be fixed. I was considering uniquely identifying the `conn` and input table to pass a `.frequency_id` argument to `rlang::warn` so that each *combination* is only warned about once per session, but after some consideration, it might be better to just always show the warning.